Publish 12/24 edition (#277)
kubernetes-sigs/lwkd: _posts/2023-12-24-update.md

---
layout: post
title: Week Ending December 24, 2023
date: 2023-01-26 22:00:00 -0000
slug: 2023-01-24-update
---

## Developer News

SIG Release [updated](https://groups.google.com/a/kubernetes.io/g/dev/c/JfZQhip-R7U) it's roadmap for 2024 onwards. If you want to contribute, just reach out to SIG Release using the K8s Slack (#sig-release) or comment on any of the open GitHub issues.

## Release Schedule

**Next Deadline: 1.30 Cycle Begins, January 2024**

We are in the holiday period between releases; dates for 1.30 have not been published yet. If you want to become a part of the [Release team](https://github.com/kubernetes/sig-release/issues/2377), apply soon!

## Featured PRs

## KEP of the Week

## Other Merges

*

## Promotions

*

## Deprecated

*

## Version Updates

*

## Subprojects and Dependency Updates

> 

Reviewer: ```suggestion

*  [cri-io/cri-io to 1.29.0](https://github.com/cri-o/cri-o/releases/tag/v1.29.0)  is released 
```

Author: @mfahlandt I was working on this right now! I will push the changes and ping you for a review

---
layout: post
title: Week Ending December 24, 2023
date: 2023-01-26 22:00:00 -0000
slug: 2023-01-24-update
---

## Developer News

> SIG Release [updated](https://groups.google.com/a/kubernetes.io/g/dev/c/JfZQhip-R7U) it's roadmap for 2024 onwards. If you want to contribute, just reach out to SIG Release using the K8s Slack (#sig-release) or comment on any of the open GitHub issues.

Reviewer: ```suggestion
SIG Release [updated](https://groups.google.com/a/kubernetes.io/g/dev/c/JfZQhip-R7U) its roadmap for 2024 onwards. If you want to contribute, just reach out to SIG Release using the K8s Slack (#sig-release) or comment on any of the open GitHub issues.
```

## Release Schedule

**Next Deadline: 1.30 Cycle Begins, January 2024**

We are in the holiday period between releases; dates for 1.30 have not been published yet. If you want to become a part of the [Release team](https://github.com/kubernetes/sig-release/issues/2377), apply soon!

> 

Reviewer: ```suggestion
Shadow applications are open until January Wednesday 3rd 2024 and will be reviewed until Friday 12th 2024. Use this [Form](https://docs.google.com/forms/d/e/1FAIpQLScTWKYMqO8TUKjqqvQzmjZ1y03mefrKqrAiPvhVqkvdzVrHAQ/viewform) to apply. 
```

Maybe we add the Shadow Form as well

## Featured PRs

> 

Reviewer 2: Could mention https://github.com/kubernetes/website/pull/44478 - I'd appreciate having that once announced.

Author: Yes, I was planning to add it today under featured PRs!

> ### [#44478: Add shortcode to generate Feature gates tables](https://github.com/kubernetes/website/pull/44478)

Reviewer 2: optionally:
```markdown
### k/website #44478: [Add shortcode to generate Feature gates tables](https://github.com/kubernetes/website/pull/44478)
```
as this is 44478 but not from the usual place.

## KEP of the Week

### [KEP 4020: Unknown Version Interoperability Proxy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/4020-unknown-version-interoperability-proxy)

> This KEP proposes to solve an issue that arises when multiple apiservers with different versions are running in a cluster. All the different apiservers might not be able to server all the resources in all the different versions. This can happen when a cluster updgrade/downgrade is going or when a runtime-config changes. This proposed filter will proxy clients to an apiserver that is capable of handling the request from the client. The KEP proposes to use the existing `StorageVersion` API to figure out which group, version and resources an apiserver can serve. If the apiserver doesn't have the requested group/version/resource locally, the request will be proxied to one of the apiservers listed in the `ServerStorageVersion` object. 

Reviewer: ```suggestion
This KEP proposes to solve an issue that arises when multiple apiservers with different versions are running in a cluster. All the different apiservers might not be able to server all the resources in all the different versions. This can happen when a cluster upgrade/downgrade is going or when a runtime-config changes. This proposed filter will proxy clients to an apiserver that is capable of handling the request from the client. The KEP proposes to use the existing `StorageVersion` API to figure out which group, version and resources an apiserver can serve. If the apiserver doesn't have the requested group/version/resource locally, the request will be proxied to one of the apiservers listed in the `ServerStorageVersion` object. 
```

> This KEP proposes to solve an issue that arises when multiple apiservers with different versions are running in a cluster. All the different apiservers might not be able to server all the resources in all the different versions. This can happen when a cluster upgrade/downgrade is going or when a runtime-config changes. This proposed filter will proxy clients to an apiserver that is capable of handling the request from the client. The KEP proposes to use the existing `StorageVersion` API to figure out which group, version and resources an apiserver can serve. If the apiserver doesn't have the requested group/version/resource locally, the request will be proxied to one of the apiservers listed in the `ServerStorageVersion` object. 

Reviewer 2: ```suggestion
This KEP proposes to solve an issue that arises when multiple API servers are running in a cluster, and specifically when they are not all serving the same set of API versions. That situaton can happen when a cluster upgrade / downgrade is happening, or during the rollout of a a runtime configuration change. All the different API servers might not be able to server all the resources in all the different versions. This proposed filter will proxy clients to an API server that is capable of handling the request from the client. The KEP proposes to use the existing `StorageVersion` mechanism to figure out which group, version and resource types an API server can serve. If the API server doesn't have the requested group/version/resource locally, the request will be proxied to one of the API servers listed in the `ServerStorageVersion` data.
```

> Major work has been done to facilitate the generation of the [feature gates reference](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) in the Kubernetes website from [information stored in feature description files](https://github.com/kubernetes/website/issues/25645). Custom Hugo shortcodes for generating feature gates tables have been added in this PR. The new `feature-gate-table` shortcode generates feature gate tables for features in Alpha or Beta, graduated or deprecated features, as well as feature gates that are removed. The shortcode also validates the name of the stage and the format of the version in the feature description file. This PR builds on top of [#41793 Switch to per-file feature gate descriptions](https://github.com/kubernetes/website/pull/41793), which added a shortcode to render feature gate descriptions. 

Reviewer 2: Looks great. Thanks.

> 

Reviewer 2: **Optionally**
```suggestion

If you're working on change that's behind a feature gate, read the updated [advice](https://kubernetes.io/docs/contribute/new-content/new-features/#ready-for-review-feature-gates) about how to document the gate or update it for a graduation.

```

This KEP was in alpha stage in the v1.28 release

## Other Merges

* Replace k8s.io/utils/integer functions with [built-in max and min functions](https://github.com/kubernetes/kubernetes/pull/122310).
* Clarification to warning around [duplicate env var entries](https://github.com/kubernetes/kubernetes/pull/122126) to mention that duplicates can be dropped when using apply.
* Clean up of [unused HealthcheckType](https://github.com/kubernetes/kubernetes/pull/122083).
* [Scheduler First Fit](https://github.com/kubernetes/kubernetes/pull/122435) has been fixed after it was [reverted](https://github.com/kubernetes/kubernetes/pull/122317) due to a flaky test.
* Scheduler configuration validation gets a [more understandable error message](https://github.com/kubernetes/kubernetes/pull/122387).
* kubectl to [initialize the customizable attachFunc](https://github.com/kubernetes/kubernetes/pull/122447) when the `--attach` flag is passed to fix a bug that was causing the initialization to be skipped.
* Update to the [help message of the interactive flag in kubectl delete](https://github.com/kubernetes/kubernetes/pull/122441).

## Deprecated

> * GA FeatureGates being removed: [APISelfSubjectReview](https://github.com/kubernetes/kubernetes/pull/122032)

Reviewer 2: ```suggestion
* GA feature gates being removed: [APISelfSubjectReview](https://github.com/kubernetes/kubernetes/pull/122032)
```

But also, this isn't a deprecation. It's a removal, which is different. Deprecated things still work, you should stop using them. Removed things aren't there for you to try using them.
